mawabot/cogs/guild.py

```diff
@@ -0,0 +1,150 @@
+#
+# cogs/guild.py
+#
+# mawabot - Maware's selfbot
+# Copyright (c) 2017 Ma-wa-re, Ammon Smith
+#
+# mawabot is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+''' Has several commands that get guild information '''
+import re
+
+import discord
+from discord.ext import commands
+
+__all__ = [
+    'setup',
+]
+
+NUMERIC_REGEX = re.compile(r'[0-9]+')
+ROLE_MENTION_REGEX = re.compile(r'<@&([0-9]+)>')
+
+class Guild:
+    __slots__ = (
+        'bot',
+    )
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    # Helper methods
+    @staticmethod
+    async def _guild_check(ctx):
```

Comment:
There is a in-built commands decorator for guild only `@commands.guild_only()`. However all it does it make it ignore the command if it's not done in a guild and raises `commands.errors.NoPrivateMessage` to the bots `on_command_error` event reference so it is up to you how you want to handle it